daemon/k8s_watcher.go

```diff
@@ -1292,24 +1302,162 @@ func (d *Daemon) deleteCiliumNetworkPolicy(obj interface{}) {
 	scopedLog.Debug("Deleting CiliumNetworkPolicy")
 
 	rules, err := ruleCpy.Parse()
-	if err != nil {
+	if err == nil {
+		if len(rules) > 0 {
+			// All rules were stored with the same labels so there's no need
+			// to delete based on all labels' rules.
+			_, err = d.PolicyDelete(rules[0].Labels)
+		}
+	}
+	if err == nil {
+		scopedLog.Info("Deleted CiliumNetworkPolicy")
+	} else {
 		scopedLog.WithError(err).Warn("Unable to delete CiliumNetworkPolicy")
+	}
+}
+
+// Deprecated: use updateCiliumNetworkPolicyV2
+func (d *Daemon) updateCiliumNetworkPolicyV1(ciliumV1Store cache.Store,
+	oldObj interface{}, newObj interface{}) {
+
+	oldRule, ok := oldObj.(*cilium_v1.CiliumNetworkPolicy)
+	if !ok {
+		log.WithField(logfields.Object+".old", logfields.Repr(oldObj)).
+			Warn("Ignoring invalid k8s CiliumNetworkPolicy modification")
+		return
+	}
+	newRules, ok := newObj.(*cilium_v1.CiliumNetworkPolicy)
+	if !ok {
+		log.WithField(logfields.Object+".new", logfields.Repr(newObj)).
+			Warn("Ignoring invalid k8s CiliumNetworkPolicy modification")
+		return
+	}
+
+	oldRuleCpy := oldRule.DeepCopy()
+	_, err := oldRuleCpy.Parse()
+	if err != nil {
+		log.WithError(err).WithField(logfields.Object, logfields.Repr(oldRuleCpy)).
+			Warn("Error parsing old CiliumNetworkPolicy rule")
+		return
+	}
+
+	newRuleCpy := newRules.DeepCopy()
+	_, err = newRules.Parse()
+	if err != nil {
+		log.WithError(err).WithField(logfields.Object, logfields.Repr(newRuleCpy)).
+			Warn("Error parsing new CiliumNetworkPolicy rule")
+		return
+	}
+
+	// Since we are updating the status map from all nodes we need to prevent
```

Comment:
I assume this comment means that we want to ignore updates if no spec change has occured

Reply:
```
	// Ignore updates of the spec remains unchanged.
```

Reply:
`if`